activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -600,7 +608,7 @@ def exec_no_cache(sql, name, binds)
           type_casted_binds = type_casted_binds(binds)
           log(sql, name, binds, type_casted_binds) do
             ActiveSupport::Dependencies.interlock.permit_concurrent_loads do
-              @connection.async_exec(sql, type_casted_binds)
+              @connection.exec_params(sql, type_casted_binds)
```

Comment:
Would exec_params be safe as a 5.2 backport? (would we need to backport to 5.1/5.0/4.2 as well?)

Reply:
also I find the semantics a bit confusing cause I would expect exec_params to work like exec and async_exec_params to work like async_exec ... but it appears that exec_params is always async.

Reply:
> Would exec_params be safe as a 5.2 backport?

Yes, `exec_params` has been there since pg-0.15.0. And starting with pg-1.1.0 it will be a alias for `async_exec_params`.

In pg < 1.1 `exec_params` did what `sync_exec_params` will do in pg-1.1. `sync_exec_params` differed from `async_exec_params`, in that it doesn't stop at signals. It delayes signal processing until the SQL statement has been finished.

Using `exec_params` instead of `async_exec` is therefore kind of regression regarding signals. But there are still other pg methods in use which delay signal processing equally and it is only valid when running with pg < 1.1, so that I don't think it's actually a problem. I asked to get the other methods fixed in PR #32820, but then we came to the conclusion, that it's better to alias all query methods to the corresponding `async_foo` method in pg-1.1 instead.